src/config/express.js

```diff
@@ -1,71 +1,85 @@
 const express = require('express')
 const router = express.Router()
 const path = require('path')
-const env = require('./env')
 const favicon = require('serve-favicon')
 const bodyParser = require('body-parser')
-const session = require('express-session')
-const cookieParser = require('cookie-parser')
 const BunyanMiddleware = require('bunyan-middleware')
 const HttpStatus = require('http-status-codes')
 
 module.exports = logger => {
-    /**
-     * Global App Config
-     */
+  /**
+   * Global App Config
+   */
   let app = express()
-
   app.use(BunyanMiddleware({
     headerName: 'X-Request-Id',
     propertyName: 'reqId',
     logName: 'req_id',
     obscureHeaders: [],
     logger: logger
   }))
+  app.use(bodyParser.json())
+  app.use(bodyParser.urlencoded({
+    extended: true
+  }))
+
+  app.use(favicon('./public/img/favicon.ico'))
+  app.use(express.static(path.join(__dirname, 'public')))
 
-  app.use(session({
-    saveUninitialized: true,
-    resave: true,
-    secret: env('SESSION_SECRET')
+  app.use(BunyanMiddleware({
+    headerName: 'X-Request-Id',
+    propertyName: 'reqId',
+    logName: 'req_id',
+    obscureHeaders: [],
+    logger: logger
   }))
 
   app.use(bodyParser.json())
-
   app.use(bodyParser.urlencoded({
     extended: true
   }))
 
-  app.use(cookieParser())
-
   app.use(favicon('./public/img/favicon.ico'))
-
   app.use(express.static(path.join(__dirname, 'public')))
 
-    /**
-     * Routing Config
-     */
+  /**
+   * Routing Config
+   */
   app.use('/api', router)
 
+  require('../routes/auth.routes')(router)
+  require('../routes/user.routes')(router)
+
   app.get('/', (request, response) => {
     response.send('Hello World!')
   })
 
-  require('../routes/user.routes')(router)
   app.use((req, res, next) => {
     let err = new Error(HttpStatus.getStatusText(HttpStatus.NOT_FOUND))
     err.status = HttpStatus.NOT_FOUND
     next(err)
   })
 
-    /**
-     * Global Error Config
-     */
+  /**
+   * CORS Config
+   */
+  app.use((req, res, next) => {
+    res.header('Access-Control-Allow-Origin', '*')
+    res.header('Access-Control-Allow-Methods', 'PUT, GET, POST, DELETE, OPTIONS')
```

Comment:
The OPTIONS method represents a request for information about the communication options available on the request/response chain identified by the Request-URI. **This method allows the client to determine the options and/or requirements associated with a resource, or the capabilities of a server, without implying a resource action or initiating a resource retrieval**.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html